Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -158,6 +177,28 @@ public class ClientsSynchronizer: BaseSingleCollectionSynchronizer, Synchronizer
         return deferResult((false, []))
     }
 
+    private func uploadRecords(should: Bool, ifUnmodifiedSince: Timestamp?, toLocalClients localClients: RemoteClientsAndTabs, withServer storageClient: Sync15CollectionClient<ClientPayload>) -> Success {
+        return localClients.getCommands() >>== { clients in
+            for client in clients {
+                storageClient.get(client.guid!) >>== { response in
+                    let record = response.value
+                    if var clientRecord = record.payload.asDictionary {
+                        clientRecord["commands"] = JSON(record.payload.commands + (client.commands?.map { JSON($0.value) } ?? []))
+                        storageClient.put(Record(id: client.guid!, payload: ClientPayload(JSON(clientRecord)), ttl: ThreeWeeksInSeconds), ifUnmodifiedSince: record.modified)
+                            >>== { resp in
+                                log.debug("Client record upload succeeded.")
+                                localClients.deleteCommands(client.guid!)
+                        }
+                    }
+                }
+            }
+            return succeed()
+        } >>== {
+            self.maybeUploadOurRecord(should, ifUnmodifiedSince: ifUnmodifiedSince, toServer: storageClient)
```

Comment:
We should upload ours first or unconditionally. If we screw up sending, we'll never upload our own record, and — worst case — we'll re-run commands we've received forever. Ruh roh.